File: src/数组/35.搜索插入位置.cpp
#include <iostream>
#include <vector>
class Solution {
public:
    int searchInsert(std::vector<int>& nums, int target)
    {
        int left = 0;
        int right = nums.size() - 1;
        while (left <= right) {
            int middle = left + (right - left) / 2;
            if (target == nums[middle]) {
                return middle;
            } else if (target > nums[middle]) {
                left = middle + 1;
            } else {
                right = middle - 1;
            }
        }
        return left;
    }
};

int main()
{
    Solution s;
    std::vector<int> nums = { 1, 3, 5, 6 };
    int target = 3;
    int res = s.searchInsert(nums, target);
    std::cout << res << std::endl;
}

File: 数组/35.搜索插入位置.py
from typing import List


class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums) - 1
        while left <= right:
            mid = left + (right - left) // 2
            if nums[mid] < target:
                left = mid + 1
            elif nums[mid] > target:
                right = mid - 1
            else:
                return mid
        return left


if __name__ == "__main__":
    nums = [1, 3, 5, 6]
    print(Solution().searchInsert(nums, 5))
